pandas/core/exchange/column.py

```diff
@@ -0,0 +1,375 @@
+        except NoBufferPresent:
+            pass
+
+        return buffers
+
+    def _get_data_buffer(
+        self,
+    ) -> Tuple[PandasBuffer, Any]:  # Any is for self.dtype tuple
+        """
+        Return the buffer containing the data and the buffer's associated dtype.
+        """
+        if self.dtype[0] in (
+            DtypeKind.INT,
+            DtypeKind.UINT,
+            DtypeKind.FLOAT,
+            DtypeKind.BOOL,
+            DtypeKind.DATETIME,
+        ):
+            buffer = PandasBuffer(self._col.to_numpy(), allow_copy=self._allow_copy)
+            dtype = self.dtype
+        elif self.dtype[0] == DtypeKind.CATEGORICAL:
+            codes = self._col.values.codes
+            buffer = PandasBuffer(codes, allow_copy=self._allow_copy)
+            dtype = self._dtype_from_pandasdtype(codes.dtype)
+        elif self.dtype[0] == DtypeKind.STRING:
+            # Marshal the strings from a NumPy object array into a byte array
+            buf = self._col.to_numpy()
+            b = bytearray()
+
+            # TODO: this for-loop is slow; can be implemented in Cython/C/C++ later
+            for obj in buf:
+                if isinstance(obj, str):
+                    b.extend(obj.encode(encoding="utf-8"))
+
+            # Convert the byte array to a Pandas "buffer" using
+            # a NumPy array as the backing store
+            buffer = PandasBuffer(np.frombuffer(b, dtype="uint8"))
+
+            # Define the dtype for the returned buffer
+            dtype = (
+                DtypeKind.STRING,
+                8,
+                ArrowCTypes.STRING,
+                Endianness.NATIVE,
+            )  # note: currently only support native endianness
+        elif self.dtype[0] == DtypeKind.DATETIME:
+            pass
+        else:
+            raise NotImplementedError(f"Data type {self._col.dtype} not handled yet")
+
+        return buffer, dtype
+
+    def _get_validity_buffer(self) -> Tuple[PandasBuffer, Any]:
+        """
+        Return the buffer containing the mask values indicating missing data and
+        the buffer's associated dtype.
+        Raises NoBufferPresent if null representation is not a bit or byte mask.
+        """
+        null, invalid = self.describe_null
+
+        if self.dtype[0] == DtypeKind.STRING:
+            # For now, use byte array as the mask.
+            # TODO: maybe store as bit array to save space?..
+            buf = self._col.to_numpy()
+            mask = []
+
+            # Determine the encoding for valid values
+            valid = 1 if invalid == 0 else 0
+
+            mask = [valid if isinstance(obj, str) else invalid for obj in buf]
+
+            # Convert the mask array to a Pandas "buffer" using
+            # a NumPy array as the backing store
+            buffer = PandasBuffer(np.asarray(mask, dtype="uint8"))
+
+            # Define the dtype of the returned buffer
+            dtype = (DtypeKind.UINT, 8, ArrowCTypes.UINT8, Endianness.NATIVE)
```

Comment:
Is there a reason/convention to encode <i>byte</i> masks as uint8 but not as bool? My concern is that sometimes it's inconvinient to work with a validity mask when it's not a boolean type, for ex. inversing doesn't work natively:
```python
>>> ~np.array([1, 1, 0, 1], dtype=np.uint8) # messy
array([254, 254, 255, 254], dtype=uint8)
>>> ~np.array([1, 1, 0, 1], dtype=np.bool) # nice
array([False, False,  True, False])
```

Reply:
good point, changed